src/cryptoadvance/specter/templates/includes/helpers.jinja

```diff
@@ -34,11 +34,15 @@
     function formatBtcAmount(valueInBTC, unitLabel, convertToSat){
         var formattedUnitLabel = formatUnitLabel(unitLabel, convertToSat);
         var formattedValue = valueInBTC;
-        if (valueInBTC) {
-            value = parseFloat(valueInBTC.toFixed(8));
+        if (formattedValue) {
             // The second condition in the if clause if True if formattedUnitLabel is equal to any element of the array
             if (convertToSat && (["Lsat", "sat", "tLsat", "tsat"].indexOf(formattedUnitLabel) > -1)) {
-                formattedValue = parseInt(value * 1e8)
+                // if it is sat, then format at integer
+                formattedValue = parseInt(parseFloat(valueInBTC) * 1e8);
+            }
+            else{
+                // else, then cut off at 8 digits
+                formattedValue = parseFloat(valueInBTC).toFixed(8);
             }
             formattedValue = `${numberWithCommas(formattedValue)}`;
```

Comment:
`numberWithCommas` only makes sense to apply to the sats amount, no? But, we are getting rid of the function anyway, so just leave it.

Reply:
I thought it also makes the commas in here 
![image](https://user-images.githubusercontent.com/60378539/196960265-5ae3addc-58bf-487f-a136-3dc1950c0f8d.png)

Reply:
Ah, true!